src/modules/telemetry-helper.js

```diff
@@ -1,29 +1,70 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 
 var clientSocket;
+var serviceToPluginMap;
+var pendingTelemetryEvents = [];
 
 function init(socket) {
     clientSocket = socket;
+    trySendPendingEvents();
 }
 
-function sendClientTelemetry(event, props, piiProps) {
+function registerPluginServices(pluginServices) {
+    serviceToPluginMap = pluginServices;
+    trySendPendingEvents();
+}
+
+function trySendPendingEvents() {
     if (!clientSocket) {
         return;
     }
 
-    var telemetryData = {
+    var unsent = [];
+
+    pendingTelemetryEvents.forEach(function (eventData) {
+        if (mustMapServiceToPlugin(eventData) && !serviceToPluginMap) {
+            unsent.push(eventData);
+        } else {
+            sendClientTelemetry(eventData);
+        }
+    });
+    
+    pendingTelemetryEvents = unsent;
+}
+
+function mustMapServiceToPlugin(eventData) {
+    return !!eventData.props.service && !eventData.plugin;
+}
+
+function sendClientTelemetry(event, props, piiProps) {
+    var eventData = {
         event: event,
         props: props,
         piiProps: piiProps
```

Comment:
Thank you for the clarification @guillaumejenkins! :) It helps a lot. I didn't know about the acronym. Maybe it could be worth to add some documentation, same as you explain to me, to help developers to understand the usages of this, and also for those who wants to contribute to the project.